lib/src/string_scanner.dart

```diff
@@ -120,6 +139,45 @@ class StringScanner {
     _fail(name);
   }
 
+  /// Consumes a single Unicode code unit and returns it.
+  ///
+  /// This works like [readChar], except that it automatically handles UTF-16
+  /// surrogate pairs. Specifically, if the next character is a high surrogate,
```

Comment:
-> ... if the next two code points are a surrogate pair, it consumes them both and returns the corresponding Unicode code point.

As written, it sounds like it consumes the code point following a high surrogate, whether it's a low surrogate or not.